pkg/templates/templates.go

```diff
@@ -352,42 +353,71 @@ func UsesEncryption(template []byte, startDelim string, stopDelim string) bool {
 	return usesEncryption
 }
 
-// getValidContext takes an input context struct with string fields and
-// validates it. If it is valid, the context will be returned as is. If the
-// input context is nil, an empty struct will be returned. If it's not valid, an
-// error will be returned.
-func getValidContext(context interface{}) (ctx interface{}, _ error) {
-	var ctxType reflect.Type
+// getValidContext takes an input context struct and validates it. If it is valid, the context will be returned as is.
+// If the input context is nil, an empty struct will be returned. If it's not valid, an error will be returned.
+func getValidContext(value interface{}) (ctx interface{}, _ error) {
+	if value == nil {
+		return struct{}{}, nil
+	}
 
-	if context == nil {
-		ctx = struct{}{}
+	// Require the context to be a struct
+	if reflect.TypeOf(value).Kind() != reflect.Struct {
+		return nil, fmt.Errorf("%w, got %s", ErrInvalidContextType, reflect.TypeOf(value))
+	}
 
-		return ctx, nil
+	// Require the context to have fields of strings or maps/structs with string/map values/fields.
+	err := getValidContextHelper(value)
+	if err != nil {
+		return nil, err
 	}
 
-	ctxType = reflect.TypeOf(context)
+	return value, nil
+}
 
-	if ctxType.Kind() != reflect.Struct {
-		return nil, fmt.Errorf("%w, got %s", ErrInvalidContextType, ctxType)
+func getValidContextHelper(value interface{}) error {
+	if value == nil {
+		return nil
 	}
 
-	for i := 0; i < ctxType.NumField(); i++ {
-		f := ctxType.Field(i)
+	f := reflect.TypeOf(value)
+
+	switch f.Kind() {
+	case reflect.String:
```

Comment:
Because of 
```
if reflect.TypeOf(value).Kind() != reflect.Struct {
		return nil, fmt.Errorf("%w, got %s", ErrInvalidContextType, reflect.TypeOf(value))
	}

```
I don't think it is reachable